### PR TITLE
Clarify scenario where AsciiExt appears to operate on non-ASCII

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes https://github.com/rust-lang/rust/issues/31203

### CODE REVIEW

File: src/libstd/ascii.rs [hunk 1]
```diff
@@ -18,6 +18,25 @@ use mem;
 use ops::Range;
 
 /// Extension methods for ASCII-subset only operations on string slices.
+///
+/// Be aware that operations on seemingly non-ASCII characters can sometimes have unexpected
+/// results. Consider this example:
+///
+/// ```
+/// # use std::ascii::AsciiExt;
```
Comment (steveklabnik) [inline src/libstd/ascii.rs]: Showing the imports is considered better nowadays.

File: src/libstd/ascii.rs [hunk 2]
```diff
@@ -18,6 +18,25 @@ use mem;
...
+/// assert_eq!("café".to_ascii_uppercase(), "CAFÉ");
+/// assert_eq!("café".to_ascii_uppercase(), "CAFé");
+/// ```
+///
+/// On the first line, the lowercased string is represented `"cafe\u{301}"` (the last character is
+/// an acute accent [combining character][]). Unlike the other characters in the string, the
```
Comment (steveklabnik) [inline src/libstd/ascii.rs]: you don't need the empty `[]`, it works without it

File: src/libstd/ascii.rs [hunk 3]
```diff
@@ -18,6 +18,25 @@ use mem;
...
+/// combining character will not get mapped to an uppercase variant, resulting in `"CAFE\u{301}"`.
+/// On the second line, the lowercased string is represented `"caf\u{e9}"` (the last character is a
+/// single unicode character representing an 'e' with an acute accent). Since the last character is
```
Comment (steveklabnik) [inline src/libstd/ascii.rs]: Unicode should be capitalized

**Comment by steveklabnik** [comment]
Thanks so much! A few nits, and also, could you wrap this to 80 cols, not 100? `libstd` uses 80.

**Comment by frewsxcv** [comment]
Comments have been addressed

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thanks so much

**Pull request merged**